src/web/crate_details.rs

```diff
@@ -91,6 +92,34 @@ pub struct Release {
     pub target_name: String,
 }
 
+#[fn_error_context::context("fetching readme for {name} {version}")]
+fn fetch_readme_from_source(
+    storage: &Storage,
+    name: &str,
+    version: &str,
+    archive_storage: bool,
+) -> anyhow::Result<String> {
+    let manifest = storage.fetch_source_file(name, version, "Cargo.toml", archive_storage)?;
+    let manifest = String::from_utf8(manifest.content)
+        .context("parsing Cargo.toml")?
+        .parse::<toml::Value>()
+        .context("parsing Cargo.toml")?;
+    let paths = match manifest.get("package").and_then(|p| p.get("readme")) {
+        Some(toml::Value::Boolean(true)) => vec!["README.md"],
+        Some(toml::Value::Boolean(false)) => vec![],
+        Some(toml::Value::String(path)) => vec![path.as_ref()],
+        _ => vec!["README.md", "README.txt", "README"],
+    };
+    for path in &paths {
+        if let Ok(readme) = storage.fetch_source_file(name, version, path, archive_storage) {
```

Comment:
we should only skip over the errors when it's an `PathNotFoundError`. Other cases should fail the function, should they?

Reply:
Yep, changed to do that.